builtin/credential/aws/path_login.go

```diff
@@ -1047,18 +1047,31 @@ func (b *backend) pathLoginRenewIam(ctx context.Context, req *logical.Request, d
 		case !roleEntry.ResolveAWSUniqueIDs && strutil.StrListContains(roleEntry.BoundIamPrincipalARNs, canonicalArn): // check 2 passed
 		default:
 			// check 3 is a bit more complex, so we do it last
+			// only try to look up full ARNs if there's a wildcard ARN in BoundIamPrincipalIDs.
+			hasWildcardBind := false
+			for _, principalARN := range roleEntry.BoundIamPrincipalARNs {
```

Comment:
Since this loop is used in a couple of places and the function is already complex, it would be good to make it into a small helper function that does this check and returns a `bool`. Then the use here and below can reduce to:

```go
if !hasWildcardBind(roleEntry.BoundIamPrincipleARNs) {
   return nil, fmt.Errorf(...)
}
```

Reply:
Good call, addressed in https://github.com/hashicorp/vault/pull/10036/commits/50fbee14ca914c23f22170b0b86b2430142655af